build_tools/teatime.py

```diff
@@ -103,6 +110,60 @@ def finish(self):
                 b"[" + self.label + b" completed in " + run_pretty.encode() + b"]\n"
             )
 
+        # Return if upload is not configured
+        if not (self.upload_to_s3 and self.s3_bucket and self.s3_subdir):
+            return
+
+        repo_root = Path(__file__).resolve().parent.parent
+        build_dir = repo_root / self.base_build_dir
+        amdgpu_family = os.getenv("AMDGPU_FAMILIES")
+
+        # Step 1: Run create_log_index.py (if AMDGPU_FAMILIES is defined)
+        if amdgpu_family:
+            try:
+                index_script = repo_root / "build_tools" / "create_log_index.py"
+                print(f"[TEATIME] Indexing logs for AMDGPU_FAMILIES={amdgpu_family}")
+                subprocess.run(
+                    [
+                        sys.executable,
+                        str(index_script),
+                        "--build-dir",
+                        str(build_dir),
+                        "--amdgpu-family",
+                        amdgpu_family,
+                    ],
+                    check=True,
+                )
+            except subprocess.CalledProcessError as e:
+                print(f"[WARN] create_log_index.py failed: {e}", file=sys.stderr)
+            except Exception as e:
+                print(
+                    f"[WARN] Unexpected error during log indexing: {e}", file=sys.stderr
+                )
+        else:
+            print("[WARN] AMDGPU_FAMILIES not set; skipping log indexing")
+
+        # Step 2: Upload logs to S3
+        try:
+            upload_script = repo_root / "build_tools" / "upload_logs_to_s3.py"
+            s3_path = f"s3://{self.s3_bucket}/{self.s3_subdir}"
+            print(f"[TEATIME] Uploading logs to {s3_path}")
+            subprocess.run(
+                [
+                    sys.executable,
+                    str(upload_script),
+                    "--build-dir",
+                    str(build_dir),
+                    "--s3-base-path",
+                    s3_path,
+                ],
+                check=True,
+            )
+        except subprocess.CalledProcessError as e:
+            print(f"[WARN] Log upload failed: {e}", file=sys.stderr)
+        except Exception as e:
+            print(f"[WARN] Unexpected error during log upload: {e}", file=sys.stderr)
```

Comment:
What are the assumption with regards to uploading logs? Is it purely optional or is it expected to work when the environment variables are set accordingly? The minimum is to document the behavior. Anyway, I would vote for the latter as it otherwise hard to catch any error and we use this script in the CI. We don't want to notice that logs are missing if we really need them. Another option would be to add a setting to control if the script should fail if the log upload did not succeeded.

Reply:
We have the environment variables below -  if they are not set -  uploading will not work
```
      TEATIME_S3_UPLOAD: "1"
      TEATIME_S3_BUCKET: "therock-artifacts"
      TEATIME_S3_SUBDIR: "${{ github.run_id }}-windows/logs/${{ inputs.amdgpu_families }}"
```

Reply:
Yes, but this only gets you a warning and wont let the script failing or did I miss something?

Reply:
I see your point - but I need to ask more questions to make it crystal.
Do we want to fail the script if a single or multiple log files cannot be uploaded?
My concern is don't we want the script continue uploading the rest of the log files even some of them missing.

Reply:
That might be an option?
```
except subprocess.CalledProcessError as e:
    print(f"[WARN] Log upload failed: {e}", file=sys.stderr)
    if os.getenv("TEATIME_FAIL_ON_UPLOAD_ERROR") == "1":
        raise
```

Reply:
Updated the code accordingly